source/common/modules/markdown-editor/hooks/note-preview.js

```diff
@@ -0,0 +1,49 @@
+const tippy = require('tippy.js').default
+const ipcRenderer = window.ipc
+
+/**
+ * A hook for displaying link tooltips which display metadata
+ * and content of a file
+ *
+ * @param   {CodeMirror}  cm  The instance to attach to
+ */
+
+module.exports = (elem) => {
+  elem.getWrapperElement().addEventListener('mousemove', (event) => {
+    let a = event.target
+
+    // Only for note links
+    if (!a.classList.contains('cm-zkn-link')) {
+      return
+    }
+
+    // If there's already a tippy on the instance, don't re-render it
+    if (a.hasOwnProperty('_tippy')) {
+      return
+    }
+    // TODO: Translate!
+
+    // Create a tippy. This will display the loading values
+    let tooltip = tippy(a, {
+      content: 'Searching For File...',
+      allowHTML: true, // Obviously
+      interactive: true,
+      placement: 'top-start', // Display at the beginning of the anchor
+      appendTo: document.body, // anchor
+      showOnCreate: true, // Immediately show the tooltip
+      arrow: false // No arrow for these tooltips
```

Comment:
Wouldn't an arrow pointing to the link contents make sense …?

Reply:
I am happy to set the arrow to true, however I should warn that that is inconsistent with the current hyperlink tooltips.

Reply:
Yes, the hyperlink tooltips don't use that. But the formatting toolbar and the footnotes tooltip use it. I think we need to focus on the question: "Should it point to something?" and I think for those note tooltips it makes sense to point back to the link!